### PR TITLE
Add Gincoin (safe code only)

### PR DESCRIPTION
**Pull request by kkabala**
Before 7 May 2018 Gincoin was using Neoscrypt instead of Lyra2Z. For now commented out that part as there are no safe Neoscrypt implementation (same way as for Mogwai).
Probably in the future I'll try to rewrite neoscrypt also.

### CODE REVIEW

File: NBitcoin.Tests/NodeBuilderEx.cs [hunk 1]
```diff
@@ -48,9 +48,10 @@ public static NodeBuilder Create([CallerMemberName] string caller = null)
 
 			//var builder = NodeBuilder.Create(NodeDownloadData.Dystem.v1_0_9_9, Altcoins.Dystem.Instance.Regtest, caller);
 
-			var builder = NodeBuilder.Create(NodeDownloadData.Bitcoin.v0_17_0, Altcoins.AltNetworkSets.Bitcoin.Regtest, caller);
+			//var builder = NodeBuilder.Create(NodeDownloadData.Bitcoin.v0_17_0, Altcoins.AltNetworkSets.Bitcoin.Regtest, caller);
 			//var builder = NodeBuilder.Create(NodeDownloadData.Liquid.v3_14_1_21, Altcoins.AltNetworkSets.Liquid.Regtest, caller);
 			//var builder = NodeBuilder.Create(NodeDownloadData.Bitcore.v0_15_2, Altcoins.Bitcore.Instance.Regtest, caller);
+			var builder = NodeBuilder.Create(NodeDownloadData.Gincoin.v1_1_0_0, Altcoins.Gincoin.Instance.Regtest, caller);
```
Comment (NicolasDorier) [inline NBitcoin.Tests/NodeBuilderEx.cs]: comment this and uncomment Bitcoin

File: NBitcoin.sln [hunk 2]
```diff
@@ -18,6 +18,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NBitcoin.Altcoins", "NBitco
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NBitcoin.TestFramework", "NBitcoin.TestFramework\NBitcoin.TestFramework.csproj", "{653C0F21-25FE-4B72-95AC-20D070A45415}"
 EndProject
```
Comment (NicolasDorier) [inline NBitcoin.sln]: remove changes in this file.

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "Submodules/Lyra2"]
```
Comment (NicolasDorier) [inline .gitmodules]: remove this

**Comment by NicolasDorier** [comment]
No submodule, duplicate the code if needed.

**Comment by kkabala** [comment]
Done. Thanks for the review Nicolas!

**Comment by eabz** [comment]
@kkabala Gincoin mainnet Magic is: 0xBD6B0CBF

**Comment by kkabala** [comment]
You're right @eabz . Already fixed, thanks!

File: NBitcoin.Altcoins/Gincoin.Internals/Consensus.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NBitcoin.Altcoins.GincoinInternals
+{
+	public static class Consensus
+	{
+		public const int LYRA2Z_TIMESTAMP = 1525651200; //7 May 2018 @ midnight UTC
```
Comment (NicolasDorier) [inline NBitcoin.Altcoins/Gincoin.Internals/Consensus.cs]: Remove this, just hardcode where you use it

File: NBitcoin.TestFramework/WellknownNodeDownloadData.cs [hunk 5]
```diff
@@ -738,6 +734,39 @@ public class ColossusNodeDownloadData
...
+					Archive = "gincoincore-{0}-windows-64bit.exe",
+					Executable = "gincoincore-{0}-windows-64bit.exe",
+					Hash = "B64E4C334D3597FC7A094DE4CA1955DBE06E9C5856598D9A2B3D8E5907D71EF9"
+				},
+				Linux = new NodeOSDownloadData()
+				{
+					DownloadLink =
+						"https://github.com/gincoin-dev/gincoin-core/releases/download/{0}.0/gincoincore-{0}-linux-64bit.tar.gz",
+					Archive = "gincoincore-{0}-linux-64bit.tar.gz",
+					Executable = "gincoincore-{0}-linux-64bit/bin/dashd",
```
Comment (NicolasDorier) [inline NBitcoin.TestFramework/WellknownNodeDownloadData.cs]: dashd?

**Comment by kkabala** [comment]
@NicolasDorier could you re-run AppVeyor build? It doesn't seem that any of my last 2 commits is able to make this unit test fail and before everything was built successfully.
On my local PC this test passes.

**Comment by kkabala** [comment]
OK, everything seems to be ok now with builds.
I've force pushed as needed to change a username of some commits

**Pull request merged**